### PR TITLE
Williamhe/1386180 add improved build warnings errors python

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1386180>.
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -211,6 +219,8 @@ fi
 			echo "'collectstatic' exited with exit code $EXIT_CODE."
 			ELAPSED_TIME=$(($SECONDS - $START_TIME))
 			echo "Done in $ELAPSED_TIME sec(s)."
+		else
```
Comment (arroyc) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:222]: else indentation

File: src/Detector/Python/PythonDetector.cs [hunk 2]
```diff
@@ -40,15 +40,36 @@ public PlatformDetectorResult Detect(DetectorContext context)
...
+                        string line = reader.ReadLine().ToLower();
+                        if (line.StartsWith("django"))
+                        {
+                            hasDjangoModule = true;
+                        }
+                    }
+                }
+                if (!hasDjangoModule)
+                {
+                    _logger.LogWarning($"Missing django module in {PythonConstants.RequirementsFileName}");
```
Comment (arroyc) [inline src/Detector/Python/PythonDetector.cs:65]: this is logging in build log .. i wonder if we need to redirect it to standard output/error as well?

File: src/Detector/Python/PythonDetector.cs [hunk 3]
```diff
@@ -40,15 +40,36 @@ public PlatformDetectorResult Detect(DetectorContext context)
...
+                }
             }
             else
             {
-                _logger.LogInformation(
-                    $"Cound not find {PythonConstants.RequirementsFileName} at the root of the repo.");
+                string errorMsg = $"Cound not find {PythonConstants.RequirementsFileName} at the root of the repo.";
```
Comment (arroyc) [inline src/Detector/Python/PythonDetector.cs:70]: same comment about this error message

**Review by arroyc — COMMENTED** [review]
we need to clarify little more in terms of logging/vs parsing from standard output/error.

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (william-msft) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:222]: `else` seems to align correctly with `fi` on line 224

File: src/Detector/Python/PythonDetector.cs (hunk 3 above)
Comment (william-msft) [reply src/Detector/Python/PythonDetector.cs:70]: Per our discussion, we'll be creating a separate method for logging. More updates to come

Comment (william-msft) [reply src/Detector/Python/PythonDetector.cs:70]: First commit of improved logging: https://github.com/microsoft/Oryx/pull/1099/commits/b25ef613569a3d4d40e0b1ece80be124903df65d

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (arroyc) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:227]: looks like there are some mix of tabs and spaces whichc is causing this indentation .. can you convert tht tabs to spaces?

Comment (william-msft) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:227]: @arroyc no spaces at the beginning of each line are displayed on my end. I'll DM you a screenshot

**Review by arroyc — CHANGES_REQUESTED** [review]

**Review by arroyc — CHANGES_REQUESTED** [review]

**Review by arroyc — APPROVED** [review]

**Pull request merged**